app/models/comment.rb

```diff
@@ -20,11 +21,13 @@ class Comment < ActiveRecord::Base
   private
 
   def create_notification
-    notification = Notification.create!(
-      user_id: article.user_id,
-      comment_id: self.id,
-      read: false
-    )
-    UserMailer.comment_create(article.user).deliver
+    if (article.user_id != self.user.id)
+      notification = Notification.create!(
+        user_id: article.user_id,
+        comment_id: self.id,
```

Comment:
also you don't need to do self. in most of this.
in ruby; there are 2 times you need to explicitly use self
1. when you're setting something you always need it. 
    EX self.name = 'Casey'. If you just do name = 'Casey' it creates a local variable and doesn't actually set it on the object.
2. if you happen to have a local variable named the same thing as an instance variable. 
EX 
```
class Person
  attr_accessor :name

  def some_function
    self.name = 'casey'
    puts name # => casey
    name = 'johnny'
    puts name # => johnny
    puts self.name # => casey

  end
end

p = Person.new
p.some_function
casey
johnny
casey


```